analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SetPropertiesInsteadOfMethods.cs

```diff
@@ -0,0 +1,20 @@
+﻿using System;
+using System.Collections.Generic;
+//using System.Collections.Immutable;
+using System.Linq;
+
+public class Program
```

Comment:
Can't we have more descriptive names dan Program and main?

Reply:
I deleted the main, since it's being built as a library (no entrypoint needed).
`Program` is just a container class to put the testcases, so I don't care about its name very much.

If you have any suggestion for it, let me know :)

Reply:
Well, is you might have seen, I prefer to split compliant and non compliant (and not applicable) code in different classes.

``` C#
class Compliant
{
    void FluentSyntax()
    {
        Set.Fluent().Fluent().Max; // Compliant
        Set.Fluent()?.Fluent().Max; // Compliant
    }
}

class NonCompliant
{
    void MinExtension()
    {
        Set.Min(); // Noncompliant
    }
}

class NotApplicatable
{
    void OtherMethodCall()
    {
        new NotApplicatable().Max(); // Compliant
    }
    
    void Max(){ }
}
```

I think it helps in telling the story about the rule.

(In cases you need (non) compliant classes/structs, using namespaces might help to achieve the same)

Reply:
It's generally better to have similar cases together. To clearly see the difference between compliant and noncompliant when they are similar.

And also when we're fixing FPs/FNs, or changing the definition of what we want to detect, we don't need to chunks of test cases around.

Reply:
Where "similar" means syntactically or thematically similar (let's say "target typed new", "loops", ReaderWriteLock class usage in locking rule, ...)

...still, generally speaking (not about this specific rule)